lib/engine/share_pool.rb

```diff
@@ -145,6 +145,15 @@ def sell_shares(bundle, allow_president_change: true, swap: nil, silent: nil)
                       swap_to_entity: swap_to_entity)
     end
 
+    def log_sell_shares(entity, verb, bundle, price, swap_text)
+      @log << "#{entity.name} #{verb} #{num_presentation(bundle)} " \
+              "of #{bundle.corporation.name} and receives #{@game.format_currency(price)}#{swap_text}"
+    end
+
+    def additional_price_adjustments(_bundle)
```

Comment:
Why does the base class need this?

Reply:
I originally duplicated the sell_shares method, as i needed to do 3 things:
send shares to the corporation
reduce broker fee,
add it to the logs.

addressing your comment, I added the 3 hooks
1- SOLD_SHARES_DESTINATION
2- additional_price_adjustments for the broker fee (keeping it generic name wise, not every game that might use this will want it to be called broker fee)
3- pulled the log into a separate method

Reply:
Okay, I missed that it was used in the base class.